backend/app/auth/supabase/supabase_exceptions.py

```diff
@@ -0,0 +1,23 @@
+from dataclasses import dataclass
+
+from supabase import AuthError, AuthInvalidCredentialsError
+
+from app.auth import AuthInvalidCredentialsException, UserException
+
+
+@dataclass(frozen=True)
+class SupabaseUserException(UserException):
```

Comment:
I don't think these are adding much either. Just have the generic AuthException that wraps the implementation-specific exception.